script/script.go

```diff
@@ -51,16 +62,29 @@ func NewLuaScript() filters.Spec {
 
 // NewLuaScriptWithOptions creates a new filter spec with options
 func NewLuaScriptWithOptions(opts LuaOptions) (filters.Spec, error) {
-	return &luaScript{opts.Modules}, nil
+	sources := opts.Sources
+	if len(sources) == 0 {
+		// backwards compatible default, allow all
+		sources = append(sources, "file", "inline")
+	} else if contains("none", opts.Sources) {
+		sources = nil
+	}
```

Comment:
I think we do not need this special casing as we check if list contains specific values

Reply:
it's better for testing errors